From: =?utf-8?q?Barnab=C3=A1s_P=C5=91cze?= <pobrn@protonmail.com>
Date: Wed, 19 Oct 2022 19:42:27 +0200
Subject: object-cache: Avoid use after free of task error

The error should not be touched after passing it to
`g_task_return_error()` as noted by its documentation:

  Note that since the task takes ownership of @error, and since the
  task may be completed before returning from g_task_return_error(),
  you cannot assume that @error is still valid after calling this.

However, previously, the code did try to free error since the
`local_error` was defined with `g_autoptr(GError)`.

(cherry picked from commit f00fd653cb7b8bc33cc84615b1c75ff878c847b0)
(cherry picked from commit c4d883ebdb707e9ac453e147829f0cec4bf949fd)
---
 shell/cc-object-storage.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/shell/cc-object-storage.c b/shell/cc-object-storage.c
index f52ba59..f9a08f3 100644
--- a/shell/cc-object-storage.c
+++ b/shell/cc-object-storage.c
@@ -91,7 +91,7 @@ create_dbus_proxy_in_thread_cb (GTask        *task,
 
   if (local_error)
     {
-      g_task_return_error (task, local_error);
+      g_task_return_error (task, g_steal_pointer (&local_error));
       return;
     }
 
